cmd/infracost/cmd_test.go

```diff
@@ -43,20 +47,29 @@ func DefaultOptions() *GoldenFileOptions {
 }
 
 func GoldenFileCommandTest(t *testing.T, testName string, args []string, testOptions *GoldenFileOptions, ctxOptions ...func(ctx *config.RunContext)) {
-	if testOptions != nil && (testOptions.RunHCL || testOptions.OnlyRunHCL) {
+	if testOptions == nil || (!testOptions.OnlyRunTerraformCLI && !testOptions.OnlyRunTerragruntCLI) {
 		t.Run("HCL", func(t *testing.T) {
-			hclArgs := make([]string, len(args)+2)
-			copy(hclArgs, args)
-			hclArgs[len(args)] = "--terraform-parse-hcl"
-			hclArgs[len(args)+1] = "true"
-
-			goldenFileCommandTest(t, testName, hclArgs, testOptions, true, ctxOptions...)
+			goldenFileCommandTest(t, testName, args, testOptions, true, ctxOptions...)
 		})
 	}
 
-	if testOptions == nil || !testOptions.OnlyRunHCL {
+	if testOptions != nil && (testOptions.RunTerraformCLI || testOptions.OnlyRunTerraformCLI) {
 		t.Run("Terraform CLI", func(t *testing.T) {
-			goldenFileCommandTest(t, testName, args, testOptions, false, ctxOptions...)
+			tfCLI := make([]string, len(args)+2)
+			copy(tfCLI, args)
+			tfCLI[len(args)] = "--project-type"
+			tfCLI[len(args)+1] = "terraform_cli"
+			goldenFileCommandTest(t, testName, tfCLI, testOptions, false, ctxOptions...)
+		})
+	}
+
+	if testOptions != nil && (testOptions.RunTerragruntCLI || testOptions.OnlyRunTerragruntCLI) {
```

Comment:
This isn't super nice, I thought about having a `ProjectTypes` option to pass to the tests but didn't want to refactor a lot of the tests just now.

Reply:
yeah agreed, it's not great, let's leave it out of the scope of this PR